devito/propagator.py

```diff
@@ -83,11 +85,14 @@ def __init__(self, name, nt, shape, spc_border=0, time_order=0,
 
         # Cache blocking and default block sizes
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
+        self.cache_block_inner = cb_inner_dim
+        if isinstance(block_size, Iterable):
+            if len(block_size) == len(shape):
                 self.block_sizes = block_size
+            elif len(block_size) == len(shape) - 1 and not self.cache_block_inner:
```

Comment:
So, if I understand you right, we add a separate interface parameter `cb_inner_dim` just enable a shorthand notation where `block_size=n` gets broadcast to `block_size=[n, n, n]`? That doesn't sound like a particularly profitable trade-off to me. I would prefer to just pass in `block_size=[n, n]`, dummy all missing inner dimensions if `len(block_size) < len(shape)` and leave the simple broadcast for square blocks to the user application or auto-tuning script. This would reduce the complexity and special-casing in the interface, and still not require much more work from the user, IMHO.